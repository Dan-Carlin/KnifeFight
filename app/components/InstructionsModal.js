import React, { useState } from "react";
import { View, StyleSheet } from "react-native";

import Button from "./Button";
import colors from "../config/colors";
import { GotItGraphic } from "../assets/buttons/actions";
import KFHowToModalOneScreen from "../screens/instructions/KFHowToModalOneScreen";
import KFHowToModalTwoScreen from "../screens/instructions/KFHowToModalTwoScreen";
import KFHowToModalThreeScreen from "../screens/instructions/KFHowToModalThreeScreen";
import Text from "./Text";

function InstructionsModal({ onConfirm }) {
  const howToScreens = [
    <KFHowToModalOneScreen
      onNextPress={() => setCurrentScreen(howToScreens[1])}
    />,
    <KFHowToModalTwoScreen
      onPreviousPress={() => setCurrentScreen(howToScreens[0])}
      onNextPress={() => setCurrentScreen(howToScreens[2])}
    />,
    <KFHowToModalThreeScreen
      onPreviousPress={() => setCurrentScreen(howToScreens[1])}
    />,
  ];

  const [currentScreen, setCurrentScreen] = useState(howToScreens[0]);

  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>How to use this app:</Text>
      </View>
      <View style={styles.bodyContainer}>{currentScreen}</View>
      <View style={styles.buttonsContainer}>
        <View style={styles.button}>
          <Button Graphic={GotItGraphic} onPress={onConfirm} />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  bodyContainer: {
    flex: 0.83,
    padding: 10,
  },
  button: {
    height: "60%",
    width: "50%",
  },
  buttonsContainer: {
    alignItems: "center",
    flex: 0.11,
    flexDirection: "row",
    justifyContent: "flex-end",
  },
  container: {
    backgroundColor: colors.dark,
    borderColor: colors.black,
    borderRadius: 20,
    height: "165%",
    overflow: "hidden",
    transform: [{ rotate: "-90deg" }],
    width: "45%",
  },
  title: {
    fontFamily: "default",
    fontSize: 40,
    textAlign: "center",
  },
  titleContainer: {
    alignItems: "center",
    backgroundColor: "#0005",
    borderTopEndRadius: 20,
    borderTopStartRadius: 20,
    borderBottomColor: colors.black,
    borderEndColor: colors.black,
    borderStartColor: colors.light,
    borderTopColor: colors.light,
    borderWidth: 2,
    flex: 0.06,
    flexDirection: "row",
    padding: 4,
  },
});

export default InstructionsModal;
