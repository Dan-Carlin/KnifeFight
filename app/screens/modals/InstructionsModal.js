/*
InstructionsModal - Modal that activates when a game is started.
*/

// External libraries
import React, { useState } from "react";
import { View } from "react-native";

// Assets
import { GotItGraphic } from "../../assets/buttons/actions";

// Components
import Button from "../../components/Button";
import KFHowToModalOneScreen from "../instructions/KFHowToModalOneScreen";
import KFHowToModalTwoScreen from "../instructions/KFHowToModalTwoScreen";
import KFHowToModalThreeScreen from "../instructions/KFHowToModalThreeScreen";
import Text from "../../components/Text";

// Resources
import sounds from "../../assets/sounds/sounds";
import strings from "../../config/strings";
import styles from "./InstructionsModalStyles";

function InstructionsModal({ onConfirm }) {
  const howToScreens = [
    <KFHowToModalOneScreen
      onNextPress={() => setCurrentScreen(howToScreens[1])}
    />,
    <KFHowToModalTwoScreen
      onPreviousPress={() => setCurrentScreen(howToScreens[0])}
      onNextPress={() => setCurrentScreen(howToScreens[2])}
    />,
    <KFHowToModalThreeScreen
      onPreviousPress={() => setCurrentScreen(howToScreens[1])}
    />,
  ];

  const [currentScreen, setCurrentScreen] = useState(howToScreens[0]);

  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{strings.modal_instructions_title}</Text>
      </View>
      <View style={styles.bodyContainer}>{currentScreen}</View>
      <View style={styles.buttonsContainer}>
        <View style={styles.button}>
          <Button
            testID={"ins_btn_gotit"}
            Graphic={GotItGraphic}
            sound={sounds.NEXT_CONFIRM}
            onPress={onConfirm}
          />
        </View>
      </View>
    </View>
  );
}

export default InstructionsModal;
