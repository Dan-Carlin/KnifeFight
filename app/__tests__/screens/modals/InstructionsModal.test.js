import React from "react";
import renderer from "react-test-renderer";

import InstructionsModal from "../../../screens/modals/InstructionsModal";

describe("Instructions Modal screen tests", () => {
  it("renders Instructions screen correctly", () => {
    const tree = renderer.create(<InstructionsModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
