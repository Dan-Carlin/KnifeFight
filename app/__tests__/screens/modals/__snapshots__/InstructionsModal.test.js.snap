// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions Modal screen tests renders Instructions screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#414042",
      "borderColor": "#000",
      "borderRadius": 20,
      "height": "165%",
      "overflow": "hidden",
      "transform": Array [
        Object {
          "rotate": "-90deg",
        },
      ],
      "width": "45%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0005",
        "borderBottomColor": "#000",
        "borderEndColor": "#000",
        "borderStartColor": "#bcbec0",
        "borderTopColor": "#bcbec0",
        "borderTopEndRadius": 20,
        "borderTopStartRadius": 20,
        "borderWidth": 2,
        "flex": 0.06,
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Avenir",
            "fontSize": 16,
            "width": "100%",
          },
          Object {
            "fontFamily": "default",
            "fontSize": 40,
            "textAlign": "center",
          },
        ]
      }
    >
      How to use this app:
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 0.83,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.85,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginEnd": 6,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              1.
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              Set down your phone on the play area in front of you as shown.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "height": "60%",
              "justifyContent": "flex-start",
              "padding": 10,
            }
          }
        >
          <SvgMock />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.15,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": "80%",
              "width": "25%",
            }
          }
        >
          <SvgMock
            style={
              Object {
                "opacity": 0.5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            1/3
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "80%",
              "width": "25%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": "100%",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <SvgMock />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.11,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "height": "60%",
          "width": "50%",
        }
      }
    >
      <SvgMock
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      />
    </View>
  </View>
</View>
`;
